src/rg3d/animation/animation.md

```diff
@@ -1 +1,88 @@
-# Animation (WIP)
\ No newline at end of file
+# Animation (WIP)
+
+ Animation blending state machine.
+
+ Machine is used to blend multiple animation as well as perform automatic "smooth transition
+ between states. Let have a quick look at simple machine graph:
+
+ ```text
+                                                  +-------------+
+                                                  |  Idle Anim  |
+                                                  +------+------+
+                                                         |
+           Walk Weight                                   |
+ +-----------+      +-------+           Walk->Idle Rule  |
+ | Walk Anim +------+       |                            |
+ +-----------+      |       |      +-------+         +---+---+
+                    | Blend |      |       +-------->+       |
+                    |       +------+ Walk  |         |  Idle |
+ +-----------+      |       |      |       +<--------+       |
+ | Aim Anim  +------+       |      +--+----+         +---+---+
+ +-----------+      +-------+         |                  ^
+           Aim Weight                 | Idle->Walk Rule  |
+                                      |                  |
+                       Walk->Run Rule |    +---------+   | Run->Idle Rule
+                                      |    |         |   |
+                                      +--->+   Run   +---+
+                                           |         |
+                                           +----+----+
+                                                |
+                                                |
+                                         +------+------+
+                                         |  Run Anim   |
+                                         +-------------+
+ ```
+
+ Here we have Walk, Idle, Run states which uses different sources of poses:
+ - Walk - is most complicated here - it uses result of blending between
+   Aim and Walk animations with different weights. This is useful if your
+   character can only walk or can walk *and* aim at the same time. Desired pose
+   determined by Walk Weight and Aim Weight parameters combination.
+ - Run and idle both directly uses animation as pose source.
+
+ There are four transitions between three states each with its own rule. Rule
+ is just Rule parameter which can have boolean value that indicates that transition
+ should be activated.
+
+ Example:
+
+ ```rust,no_run,ignore
```

Comment:
That's cheating 😃 . Just add `# extern crate rg3d` to make it compile.

Reply:
:)